src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -161,6 +162,18 @@ public IFullNode Build()
 
             this.Services = this.BuildServices();
 
+            if (this.NodeSettings.PrintHelpAndExit)
+            {
+                foreach (var featureRegistration in this.Features.FeatureRegistrations)
+                {
+                    MethodInfo printHelp = featureRegistration.FeatureType.GetMethod("PrintHelp", BindingFlags.Public | BindingFlags.Static);
```

Comment:
If the PrintHelp methods weren't static, you'd do featureRegistration.PrintHelp(this.NodeSettings.Network), which a bit more straightforward.

Reply:
See above.